File: app/src/main/java/ys/moire/presentation/ui/view_parts/type/AbstractType.java
package ys.moire.presentation.ui.view_parts.type;

/**
 * geometry abstract class.
 */
public abstract class AbstractType {

    abstract void autoMove(final float dx);

    abstract void checkOutOfRange(final int layoutWidth, final int slope);

    abstract void addTouchVal(final int touchDx, final int touchDy);
}

File: app/src/main/java/ys/moire/presentation/ui/view_parts/type/AbstractType.kt
package ys.moire.presentation.ui.view_parts.type

/**
 * geometry abstract class.
 */
abstract class AbstractType {

    internal abstract fun autoMove(dx: Float)

    internal abstract fun checkOutOfRange(layoutWidth: Int, slope: Int)

    internal abstract fun addTouchVal(touchDx: Int, touchDy: Int)
}
